### DISCUSSION TITLE
How To: VEDirect command

### DISCUSSION THREAD

**Discussion by jacques0kruger** [jblance/mpp-solar#119] [Q&A]
What is the popper use for reading data via vedirect protocol as the below produces a error?

mpp-solar -p /dev/ttyUSB0 -P VED -c vedtext

Error: WARNING:SerialIO:send_and_receive: Serial read error: unicode strings are not supported, please encode to bytes: 'VEDTEXT'

**Comment by jblance** [comment]
need to specify vserial porttype for VED, i.e.
 `mpp-solar -p /dev/ttyUSB0 -P VED -c vedtext --porttype vserial`

add `-I` to the above for more info if things are still not working and post all the output

**Reply by jacques0kruger** [reply]
Thanks, works great using the below.

mpp-solar -p /dev/ttyUSB0 -P VED -c vedtext --porttype vserial -b 19200

**Reply by jblance** [reply]
great to hear

**Comment by jblance** [comment]
`mpp-solar -p /dev/ttyUSB0 -P VED -c vedtext --porttype vserial -b 19200`
